concordium-smart-contract-testing/src/impls.rs

```diff
@@ -1056,11 +1223,277 @@ impl Chain {
+
+        Ok(())
+    }
+
+    /// Set up a connection to an external Concordium node.
+    ///
+    /// The connection can be used for getting the current exchange rates
+    /// between CCD, Euro and Energy.
+    fn setup_external_node_connection(
+        &mut self,
+        endpoint: &str,
+    ) -> Result<(), SetupExternalNodeError> {
+        let endpoint: sdk::v2::Endpoint = FromStr::from_str(endpoint)?;
+        // Create the Tokio runtime. This should never fail, unless nested runtimes are
+        // created.
+        let runtime = runtime::Builder::new_multi_thread()
+            // Enable time, so timeouts can be used.
+            .enable_time()
+            // Enable I/O, so networking and other types of calls are possible.
+            .enable_io()
+            .thread_keep_alive(Duration::from_secs(2))
+            .build()
+            .expect("Internal error: Could not create Tokio runtime.");
+
+        // A future for getting the client. Executed below.
+        let get_client = async {
+            // Try to create the client, which also checks that the connection is valid.
+            let client = sdk::v2::Client::new(endpoint).await?;
+            Ok::<sdk::v2::Client, SetupExternalNodeError>(client)
+        };
+
+        // Get the client synchronously by blocking until the async returns.
+        let client = runtime.block_on(async {
+            timeout(EXTERNAL_NODE_TIMEOUT_DURATION, get_client)
+                .await
+                .map_err(|_| SetupExternalNodeError::Timeout)?
+        })?;
+
+        // Set or replace the node connection.
+        self.external_node_connection = Some(ExternalNodeConnection {
+            client,
+            runtime,
+            sticky_block: None,
+        });
+
+        Ok(())
+    }
+
+    /// Try to get the [`ExternalNodeConnection`] or return an error.
+    ///
+    /// The connection is only available, if the [`Chain`] has been created with
+    fn external_node_connection(
+        &mut self,
+    ) -> Result<&mut ExternalNodeConnection, ExternalNodeNotConfigured> {
+        match &mut self.external_node_connection {
+            None => Err(ExternalNodeNotConfigured),
+            Some(data) => Ok(data),
+        }
+    }
+
     /// Return the current microCCD per euro exchange rate.
     pub fn micro_ccd_per_euro(&self) -> ExchangeRate { self.parameters.micro_ccd_per_euro }
 
     /// Return the current euro per energy exchange rate.
     pub fn euro_per_energy(&self) -> ExchangeRate { self.parameters.euro_per_energy }
+
+    /// Return the current block time.
+    pub fn block_time(&self) -> Timestamp { self.parameters.block_time }
+
+    /// Return the "sticky block", i.e. the block that is saved and will be used
+    /// by default for future queries.
+    ///
+    /// See [`set_block_time_via_external_node`][Chain::set_block_time_via_external_node] for a more detailed explanation.
+    pub fn sticky_block(&self) -> Option<BlockHash> {
```

Comment:
I find it a bit odd to call this "sticky". I'm trying to think if I could guess as to what it is if I did not know, but I cannot imagine doing so.

Have you consider something more like `query_block`?

Reply:
Yes, added now.